src/DiagnosticTools.jl

```diff
@@ -42,3 +42,32 @@ function compute_total_mass!(w,L2MM,hh)
   mul!(w,L2MM,hh)
   sum(w)
 end
+
+"""
+  Full diagnostics for the shallow water equations (mass, vorticity, kinetic energy, potential energy, power)
+"""
+function compute_diagnostics_shallow_water!(model, order, Ω, dΩ, dω, qₖ, wₖ, U, V, R, S, L2MM, H1MM, g, h, u, ϕ, F, mass, vort, kin, pot, pow, step, do_print)
+  mass_i = sum(L2MM*Gridap.FESpaces.get_free_dof_values(h))
+  w      = diagnose_vorticity(model, order, Ω, qₖ, wₖ, R, S, U, V, H1MM, u)
+  vort_i = sum(H1MM*Gridap.FESpaces.get_free_dof_values(w))
+  kin_i  = 0.5*sum(∫(h*(u⋅u))dΩ)
```

Comment:
Dont we have these functions already implemented in `DiagnosticTools.jl` ... if not ... add whatever it is needed there ...

Reply:
Have you addressed this comment?

Reply:
Thanks for spotting. I have generalised the `E_k` function so that this can take h as a FEFunction (so that it can be used to compute KE for the shallow water equations)